snap/device.go

```diff
@@ -33,6 +33,10 @@ type Device interface {
 	Gadget() string
 
 	HasModeenv() bool
+	IsCoreLegacy() bool     // true if UC16/18
+	IsCoreBoot() bool       // true if UC or classic with modes
+	IsClassicBoot() bool    // true if classic with classic initramfs
+	IsClassicModeenv() bool // true if classic with core initramfs
```

Comment:
this one seems only involved in deciding whether we should reboot on base, I wonder if instead we should have a IsBootSnap(dev, SnapType) in boot?

Reply:
yes, but again there is nothing deep in these helpers, they are only useful added to Device if we have  a lot of use cases, because of the cost of mocking them correctly. So we would isolate this to that helper in boot if that's all this is neeed for.

Reply:
I've created a IsBootSnap(dev, SnapType) function now